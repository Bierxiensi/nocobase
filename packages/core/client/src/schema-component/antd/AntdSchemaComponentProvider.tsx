import * as components from '.';
import { Plugin } from '../../application-v2/Plugin';
import * as common from '../common';
import { useFilterActionProps } from './filter/useFilterActionProps';
import { requestChartData } from './g2plot/requestChartData';

export class AntdSchemaComponentPlugin extends Plugin {
  async load() {
    this.addComponents();
    this.addScopes();
  }

  addComponents() {
    this.app.addComponents({
      ...(components as any),
      ...common,
    });
  }

  addScopes() {
    this.app.addScopes({
      requestChartData,
      useFilterActionProps,
    });
  }
}
