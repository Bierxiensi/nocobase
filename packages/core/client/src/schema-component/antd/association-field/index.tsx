import { AssociationField } from './AssociationField';
import { AssociationFieldAddNewer } from './AssociationField.AddNewer';
import { AssociationFieldSelector } from './AssociationField.Selector';
import { AssociationFieldViewer } from './AssociationField.Viewer';
import { AssociationFieldNesterDecorator } from './AssociationField.Nester.Decorator';
import { Nester } from './AssociationField.Nester';
AssociationField.AddNewer = AssociationFieldAddNewer;
AssociationField.Selector = AssociationFieldSelector;
AssociationField.Viewer = AssociationFieldViewer;
AssociationField.Nester = Nester;
AssociationField.Nester.Decorator = AssociationFieldNesterDecorator;

export * from './components/SubForm';
export * from './components/TableField';

export { AssociationField };
