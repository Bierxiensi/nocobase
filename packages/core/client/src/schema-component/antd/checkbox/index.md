---
group:
  title: Schema Components
  order: 3
---

# Checkbox

## Examples

### 勾选

<code src="./demos/demo1.tsx"></code>

### 组

<code src="./demos/demo2.tsx"></code>
