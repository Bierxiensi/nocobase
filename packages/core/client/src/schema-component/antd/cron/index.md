---
nav:
  path: /client
group:
  path: /schema-components
---

# Cron

## Example

#### Cron

<code src="./demos/demo1.tsx"></code>

#### CronSet

<code src="./demos/demo2.tsx"></code>
