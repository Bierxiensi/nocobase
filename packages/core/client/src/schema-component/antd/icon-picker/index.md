---
group:
  title: Schema Components
  order: 3
---

# IconPicker

## Examples

### IconPicker

<code src="./demos/demo1.tsx"></code>
