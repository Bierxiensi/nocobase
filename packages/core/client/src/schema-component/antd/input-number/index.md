---
group:
  title: Schema Components
  order: 3
---

# InputNumber

## Examples

### InputNumber

<code src="./demos/demo1.tsx"></code>

### addonBefore/addonAfter

<code src="./demos/demo2.tsx"></code>

### High precision decimals

<code src="./demos/demo3.tsx"></code>
