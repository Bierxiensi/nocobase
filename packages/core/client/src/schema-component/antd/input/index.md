---
group:
  title: Schema Components
  order: 3
---

# Input

## Nodes

- Input
- Input.TextArea
- Input.URL

## Examples

### Input

<code src="./demos/demo1.tsx"></code>

### Textarea

<code src="./demos/demo2.tsx"></code>

### URL

<code src="./demos/demo3.tsx"></code>

### JSON

<code src="./demos/demo4.tsx"></code>
