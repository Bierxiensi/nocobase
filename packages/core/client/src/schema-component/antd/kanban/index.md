---
group:
  title: Schema Components
  order: 3
---

# Kanban

## 示例

<code src="./demos/demo1.tsx"></code>
