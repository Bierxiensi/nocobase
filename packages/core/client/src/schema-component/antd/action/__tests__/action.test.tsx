import { render, screen, fireEvent, act } from '@testing-library/react';
import { FormItem, Input } from '@formily/antd';
import { ISchema, observer, useForm } from '@formily/react';
import { Form, SchemaComponent, SchemaComponentProvider, useActionContext, ActionContext } from '@nocobase/client';
import { Action } from '../Action';
import React, { useState } from 'react';
import { ActionLink } from '../Action.Link';
import { exp } from 'mathjs';

const useCloseAction = () => {
  const { setVisible } = useActionContext();
  const form = useForm();
  return {
    async run() {
      setVisible(false);
      form.submit((values) => {
        console.log(values);
      });
    },
  };
};

const schema: ISchema = {
  type: 'object',
  properties: {
    action1: {
      'x-component': 'Action',
      'x-component-props': {
        type: 'primary',
      },
      type: 'void',
      title: 'Open',
      properties: {
        drawer1: {
          'x-component': 'Action.Drawer',
          type: 'void',
          title: 'Drawer Title',
          properties: {
            hello1: {
              'x-content': 'Hello',
              title: 'T1',
            },
            footer1: {
              'x-component': 'Action.Drawer.Footer',
              type: 'void',
              properties: {
                close1: {
                  title: 'Close',
                  'x-component': 'Action',
                  'x-component-props': {
                    useAction: '{{ useCloseAction }}',
                  },
                },
              },
            },
          },
        },
      },
    },
  },
};

const ComponentA = observer(() => {
  return (
    <SchemaComponentProvider scope={{ useCloseAction }} components={{ Form, Action, Input, FormItem }}>
      <SchemaComponent schema={schema} />
    </SchemaComponentProvider>
  );
});

const ComponentB = observer(() => {
  return (
    <SchemaComponentProvider scope={{ useCloseAction }} components={{ Form, Action, Input, FormItem }}>
      <SchemaComponent
        schema={{
          type: 'object',
          properties: {
            action1: {
              'x-component': 'Action',
              'x-component-props': {
                type: 'primary',
                size: 'large',
              },
              type: 'void',
              title: 'action',
            },
          },
        }}
      />
    </SchemaComponentProvider>
  );
});

const ComponentC = observer(() => {
  return (
    <SchemaComponentProvider scope={{ useCloseAction }} components={{ Form, ActionLink, Input, FormItem }}>
      <SchemaComponent
        schema={{
          type: 'object',
          properties: {
            action1: {
              'x-component': 'ActionLink',
              'x-component-props': {
                type: 'primary',
                size: 'large',
              },
              type: 'void',
              title: 'link',
            },
          },
        }}
      />
    </SchemaComponentProvider>
  );
});

const setup = () => {
  const returnVal = {};

  const UseComponentTest = () => {
    const [visible, setVisible] = useState(false);

    Object.assign(returnVal, {
      visible,
      setVisible,
    });

    return (
      <SchemaComponentProvider components={{ Form, Action, Input, FormItem }}>
        <ActionContext.Provider value={{ visible, setVisible }}>
          <a onClick={() => setVisible(true)}>Open</a>
          <SchemaComponent scope={{ useCloseAction }} schema={schema} />
        </ActionContext.Provider>
      </SchemaComponentProvider>
    );
  };

  render(<UseComponentTest />);

  return returnVal;
};

describe('action snapshots', () => {
  it('action ', () => {
    const { container } = render(<ComponentA />);
    expect(container).toMatchSnapshot();
  });
  it('action display by props ', () => {
    const wrapper = render(<ComponentB />);
    const element = wrapper.getByText('action');
    expect(element).toBeInTheDocument();
    expect(element.parentNode).toHaveClass('ant-btn-primary');
    expect(element.parentNode).toHaveClass('ant-btn-lg');
  });
  it('action display as link ', () => {
    const wrapper = render(<ComponentC />);
    const element = wrapper.getByText('link');
    expect(element).toBeInTheDocument();
    expect(element.tagName).toEqual('A');
    expect(element).toHaveClass('nb-action-link');
  });
  it('action visible ', () => {
    const useComponentData: any = setup();
    const btn = screen.getAllByText('Open');
    fireEvent.click(btn[0]);
    expect(useComponentData.visible).toBe(true);
  });
});
