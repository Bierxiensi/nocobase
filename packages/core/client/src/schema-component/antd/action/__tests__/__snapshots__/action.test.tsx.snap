// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action snapshots action  1`] = `
<div>
  <button
    class="ant-btn ant-btn-primary css-1xsefc2"
    type="button"
  >
    <span>
      Open
    </span>
  </button>
</div>
`;
